FilmorateApp: initial commit

AleksandrTimofeev1987 opened the pull request.

AleksandrTimofeev1987 committed:
FilmorateApp: initial commit

avfyodorov commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,26 @@
+    private String email;
+
+    @NotBlank (message = "Логин пользователя не может быть пустым.")
+    private String login;
```
Сюда хорошо бы добавить аннотацию.
@pattern(regexp = "^\\\S*$") 
нужно, чтобы пробелов вообще не было.

avfyodorov commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -0,0 +1,74 @@
+
+    // Получить список всех фильмов
+    @GetMapping
+    public List<Film> getAllFilms() {
```
Суффиксы (Фильм, Пользователь) в названии методов можно спокойно убрать. И так понятно к какому типу объекта относится метод.

avfyodorov commented on src/test/java/ru/yandex/practicum/filmorate/controller/FilmControllerTest.java:
```diff
@@ -0,0 +1,278 @@
+    private static final Film VALID_FILM = new Film(1, "film", RandomString.make(200), RELEASE_DATE, 1);
+
+    @Autowired
+    private MockMvc mockMvc;
```
По-моему, этого вы ещё не проходили. Очень хорошо, что самостоятельно  разобрались!

avfyodorov reviewed (commented):
Добрый вечер!
Работа получилась очень хорошей,  ошибок я не нашёл. Так,  пара мелких замечаний.

AleksandrTimofeev1987 committed:
FilmorateApp: commit after first review
refactor: добавить @Regex валидацию логина в классе User
refactor: использовать DirtiesContext для сброса контекста в тестах
style: переименовать тестовые методы по форме /given //when //then
style: добавить указания /given //when //then в тестовых методах
style: переименовать методы UserController и FilmController, убрав суффиксы film и user

avfyodorov reviewed (commented):
Добрый день!

Замечаний нет.
Работа принята.
